### ISSUE TITLE
bundle: false only includes entrypoint modules in output

### ISSUE THREAD

**Issue by sentience** [egoist/tsup#1000]
Minimal reproduction: https://github.com/sentience/tsup-bundle-false-issue

```ts
export default defineConfig({
  bundle: false,
  format: ["esm"],
  dts: true,
  splitting: true,
  entry: {
    index: "src/index.ts",
  },
})
```

The `bundle` option is only documented as "Disable bundling" in the API docs, but what I (perhaps incorrectly) expected it to do was to output all modules referenced by my entry module(s), but not bundle them into the entry modules.

So if I had:

**src/index.ts**
```ts
export { foo } from './foo.js'
```

**src/foo.ts**
```ts
export function foo() {
  console.log("This is a function that should make it into the output of this package.")
}
```

I would expect tsup to output two JavaScript files:

**dist/index.mjs**
```js
import { foo } from "./foo.js"
export {
  foo
}
```

**dist/foo.mjs**
```
export function foo() {
  console.log("This is a function that should make it into the output of this package.")
}
```

Instead, tsup only outputs the first of those two files. The implementation of `foo` is left out of the build output entirely.

Is this really what `bundle: false` is supposed to do? If so, I don't understand what the use case for this option is.

My use case is to publish an ESM package with no bundling in order to maximise its tree-shakeability, assuming the consuming project will use a bundler to tree-shake dependencies and produce browser-ready bundles.

**Comment by sentience** [comment]
@smeijer I would call that a work-around. If `bundle: false` produces broken output unless you declare every one of your source files as an entrypoint, that seems like a bug to me. (Or at least missing documentation.)

**Comment by marbemac** [comment]
> I would call that a work-around. If bundle: false produces broken output unless you declare every one of your source files as an entrypoint, that seems like a bug to me

Agree! What I expected when I ran into this was that the resulting output would include my specified entry files along with any other files in their module graphs (so things like test files, etc, are automatically not included in the output).

**Comment by siaikin** [comment]
It seems this issue has been around for a while. I migrated to tsdown to work around it, and they have a migration guide from tsup that was very convenient for me.

https://tsdown.dev/guide/migrate-from-tsup
